sql/sql_tablespace.cc

```diff
@@ -470,7 +470,12 @@ bool Sql_cmd_create_tablespace::execute(THD *thd) {
     encrypt_tablespace = dd::is_encrypted(m_options->encryption);
     encrypt_type = dd::make_string_type(m_options->encryption);
   } else {
-    encrypt_tablespace = thd->variables.default_table_encryption;
+    encrypt_tablespace =
+        thd->variables.default_table_encryption == DEFAULT_TABLE_ENC_ON ||
+        thd->variables.default_table_encryption ==
+            DEFAULT_TABLE_ENC_KEYRING_ON ||
+        thd->variables.default_table_encryption ==
+            DEFAULT_TABLE_ENC_ONLINE_TO_KEYRING;
```

Comment:
Is this a different bug? handled in this PR?

Reply:
yes. Basically the sync between global and session scope for d_t_e still needs some work. I have added PS-6759: Sync global value of default_table_encryption with session value of this variable to address this issue.

Reply:
Ok, will you rever this change here( and handle in PS-6759) or it is necessary for this PR?

Reply:
I have added the minimal change here - just to instruct the encryption of tablespaces based on d_t_e. Previously encrypt_tablespace = thd->variables.default_table_encryption always evaluated to true for values different than 'N'. So for ONLINE_FROM_KEYRING_TO_UNENCRYPTED the old value of encrypt_tablespace was true - and should be false.

Reply:
I had a test failing because of that.